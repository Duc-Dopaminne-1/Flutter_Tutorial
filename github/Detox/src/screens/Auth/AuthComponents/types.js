import {GENDER_ARRAY, IDENTIFY_TYPE} from '../../../assets/constants';

export const nationalIdDefault = '00000000000';
const defaultNameCity = 'Hồ Chí Minh';
export const initialState = {
  name: '',
  middleLastName: '',
  mobilePhone: '',
  email: '',
  password: '',
  confirmPassword: '',
  dob: '',
  gender: GENDER_ARRAY[0].value,
  inviteCode: '',
  campaignCode: '',
  nationalIdType: IDENTIFY_TYPE[0].value,
  nationalId: nationalIdDefault,
  nationalIdIssueDate: '2000-01-01T00:00:00.000Z',
  nationalIdIssuePlace: defaultNameCity,
  permanentAddress: {
    city: {id: 1, name: defaultNameCity, checked: true},
    district: {id: 3, name: 'Bình Thạnh', checked: true},
    ward: {id: 40, name: 'Phường 26', checked: true},
    street: 'Richmond City, 207c Nguyễn Xí',
  },
  contactAddress: {city: {}, district: {}, ward: {}, street: ''},
  isSameAddress: true,
  workingAreas: [
    {
      city: {
        id: 4,
        name: 'Bình Dương',
      },
      district: {
        id: 66,
        name: 'Dĩ An',
      },
      id: 0,
    },
    {
      city: {
        id: 1,
        name: 'Hồ Chí Minh',
      },
      district: {
        id: 24,
        name: 'Thành phố Thủ Đức',
      },
      id: 1,
    },
    {
      city: {
        id: 25,
        name: 'Bình Định',
      },
      district: {
        id: 343,
        name: 'Quy Nhơn',
      },
      id: 2,
    },
  ],
  preferPropertyTypes: [
    {
      checked: true,
      description: 'Căn hộ',
      id: '2e2b0611-e1fc-4406-b13f-8afafff1c675',
      name: 'apartment',
    },
    {
      checked: true,
      description: 'Biệt thự',
      id: '48cceac6-e202-494d-9b06-2374042f1044',
      name: 'villa',
    },
    {
      checked: true,
      description: 'Nhà phố',
      id: 'b2ba5130-faa5-458c-aba8-58c505ac59b9',
      name: 'house',
    },
    {
      checked: true,
      description: 'Đất nền',
      id: '82cc71b0-704b-470c-bca8-5f967c0e11b7',
      name: 'land',
    },
  ],
  preferPropertyPriceFrom: 0,
  preferPropertyPriceTo: 50,
  isAgree: false,
};

export type State = typeof initialState;

export type InputViewsError = {
  step1: {
    name?: String,
    middleLastName?: String,
    email?: String,
    password?: String,
    confirmPassword?: String,
    dob?: String,
  },
  step2: {
    errArea?: String,
    errProperty?: String,
    errTeam?: String,
    errNationId?: String,
    errContactAddress?: String,
    errAddress?: String,
    errReferral?: String,
    errNationalIdIssueDate?: String,
    errNationalIdIssuePlace?: String,
    errNationalIdType?: String,
  },
};

export type InputViewsProps = {
  mobilePhone: String,
  inviteCodeLink: String,
  data: State,
  validates: InputViewsError,
  dispatch: () => {},
  setName: () => {},
  setMiddleLastName: () => {},
  setPassword: () => {},
  setDob: () => {},
  onChangeGender: () => {},
  setConfirmPassword: () => {},
  setInviteCode: () => {},
  setCampaignCode: () => {},
  onChangeIdentificationType: () => {},
  onIdTextChanged: () => {},
  nationalIdIssueDate: any,
  errNationalIdIssueDate: any,
  onChangeIdentifyDate: () => {},
  onChangeIdentifyPlace: () => {},
  setInviteCode: () => {},
  permanentAddress: {street: any},
  onAddressTextChanged: () => {},
  onContactAddressTextChanged: () => {},
  onCheckSameAddress: () => {},
  contactCities: [],
  workingAreas: [],
  errArea: any,
  onRemoveArea: () => {},
  preferPropertyPriceFrom: any,
  preferPropertyPriceTo: any,
  onPriceInterestedChanged: () => {},
  addressCities: [],
  areaCities: [],
};
