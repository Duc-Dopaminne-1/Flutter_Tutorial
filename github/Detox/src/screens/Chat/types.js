/* eslint-disable sonarjs/no-duplicate-string */

import moment from 'moment';

import {translate} from '../../assets/localize';
import {getCalendarDateTimeString, unixTimestampToTimeAmPm} from '../../utils/TimerCommon';

export type IConversation = {
  unreadCount: Number,
};

type RoleName = 'member' | 'agent' | 'admin';

export type IParticipant = {
  userId: String,
  name: String,
  avatar: String,
  isAgent: Boolean,
  isMember: Boolean,
  agentId?: String,
  roleNames: RoleName[],
};

export const MESSAGE_TYPE = {
  TEXT: 1,
  PHOTO: 2,
  VIDEO: 3,
  AUDIO: 4,
  FILE: 5,
  LINK: 6,
  CONVERSATION_CREATION_NOTIFICATION: 7,
  CONVERSATION_RENAME_NOTIFICATION: 8,
  LOCATION: 9,
  CONTACT: 10,
  STICKER: 11,
  NOTIFICATION: 100,
};

export const MESSAGE_STATE = {
  INITIALIZE: 0, //The message is initialized
  SENDING: 1, //The message is sent by the sender
  SENT: 2, //The message is sent to Stringee server
  DELIVERED: 3, //The message is delivered to the recipient
  READ: 4, //The message is read by the recipient
};

export type IChatMessage = {
  _id: String,
  text: String,
  createdAt: Date,
  image?: String,
  imageUrl?: String,
  imageRatio?: Number,
  user: {
    _id: String,
    name: String,
    avatar: String,
  },
  conversationId: String,
  fileUrl?: String,
  fileLength?: Number,
};

export const parseConversation = (value, currentUserId) => {
  const id = value.id;
  const isGroup = value.isGroup;
  const {userId, name, avatar} = value.participants
    .filter(participant => participant.userId !== currentUserId)
    .map(item => ({...item, name: item.name || item.userId}))[0];
  const rawDateTime = value.lastMessage.createdAt;
  const date = convertTimeToTimeAgo(rawDateTime);
  const lastMessage = value.lastMessage.content?.content || '--';
  const unreadCount = value.unreadCount;
  const participants = value.participants;
  return {
    id,
    userId,
    name,
    isGroup,
    groupName: value.name,
    avatar,
    unreadCount,
    lastMessage,
    lastMessageType: value.lastMessage.type,
    rawDateTime,
    date,
    participants,
  };
};

export const parseMessages = (data: typeof getLastMessages, participants) => {
  const result = data.map(value => {
    const message: IChatMessage = {
      _id: value.id,
      text: value.content?.content,
      createdAt: value.createdAt,
      sequence: value.sequence,
      user: getUserFromParticipants(participants, value.sender),
      conversationId: value.conversationId,
    };
    if (value.type === MESSAGE_TYPE.PHOTO) {
      message.image = value.content?.photo?.filePath;
    }
    if (value.type === MESSAGE_TYPE.FILE) {
      message.text = value.content?.file?.filename;
      message.fileUrl = value.content?.file?.filePath;
    }
    return message;
  });
  return result;
};

export const getMessageContentByType = (type, content) => {
  switch (type) {
    case MESSAGE_TYPE.PHOTO:
      return translate('chat.type.photo');
    case MESSAGE_TYPE.FILE:
      return translate('chat.type.file');
    default:
      return content;
  }
};

export const getUserFromParticipants = (participants, idOrName) => {
  const {
    userId: _id,
    name,
    avatar,
  } = participants.filter(value => value.userId === idOrName || value.name === idOrName)[0];
  return {_id, name, avatar};
};

export const convertTimeToTimeAgo = timestamp => {
  const diff = moment(new Date()).diff(moment(timestamp), 'day');
  if (diff === 0) {
    return unixTimestampToTimeAmPm(timestamp);
  }
  return getCalendarDateTimeString(timestamp);
};

export const createConversation = {
  id: 'conv-vn-1-8AT5T0P0ET-1634524961033',
  name: 'btncall_0000120493',
  isGroup: false,
  updatedAt: 1635146158214,
  creator: 'btncall_0000123452',
  created: 1635146158211,
  unreadCount: 0,
  participants: [
    {userId: 'btncall_0000123452', name: '', avatar: ''},
    {userId: 'btncall_0000120493', name: '', avatar: ''},
  ],
  lastMessage: {
    localId: null,
    id: 'msg-vn-1-8AT5T0P0ET-1634524963617',
    conversationId: 'conv-vn-1-8AT5T0P0ET-1634524961033',
    sender: '',
    createdAt: 1635146158214,
    state: 4,
    sequence: 1,
    type: 7,
    content: {
      participants: ['btncall_0000120493', 'btncall_0000123452'],
      creator: 'btncall_0000123452',
      groupName: 'btncall_0000120493',
    },
  },
};

export const getLastConversations = [
  {
    id: 'conv-vn-1-8AT5T0P0ET-1634524961048',
    name: 'binh020304',
    isGroup: true,
    updatedAt: 1635155773086,
    creator: 'btncall_0000123452',
    created: 1635155265332,
    unreadCount: 6,
    participants: [
      {userId: 'btncall_0000123452', name: '', avatar: ''},
      {userId: 'btncall_0000120493', name: '', avatar: ''},
      {userId: 'btncall_0000123404', name: '', avatar: ''},
    ],
    lastMessage: {
      localId: null,
      id: 'msg-vn-1-8AT5T0P0ET-1634524965643',
      conversationId: 'conv-vn-1-8AT5T0P0ET-1634524961048',
      sender: 'btncall_0000120493',
      createdAt: 1635155773086,
      state: 3,
      sequence: 11,
      type: 1,
      content: {messageType: 1, text: '10', content: '10'},
    },
  },
  {
    id: 'conv-vn-1-8AT5T0P0ET-1634524961033',
    name: 'btncall_0000120493',
    isGroup: false,
    updatedAt: 1635155484063,
    creator: 'btncall_0000123452',
    created: 1635146158211,
    unreadCount: 25,
    participants: [
      {userId: 'btncall_0000123452', name: '', avatar: ''},
      {userId: 'btncall_0000120493', name: '', avatar: ''},
    ],
    lastMessage: {
      localId: null,
      id: 'msg-vn-1-8AT5T0P0ET-1634524965466',
      conversationId: 'conv-vn-1-8AT5T0P0ET-1634524961033',
      sender: 'btncall_0000123452',
      createdAt: 1635155484063,
      state: 4,
      sequence: 62,
      type: 1,
      content: {metadata: {}, content: '57'},
    },
  },
];

const getLastMessages = [
  {
    localId: 'iOS-6B0F08E0-61B6-4009-BDBD-6D8A7769925C-1635150215340361',
    id: 'msg-vn-1-8AT5T0P0ET-1634524963650',
    conversationId: 'conv-vn-1-8AT5T0P0ET-1634524961033',
    sender: 'btncall_0000123452',
    createdAt: 1635150215454,
    state: 4,
    sequence: 2,
    type: 1,
    content: {content: 'i is a little bit'},
  },
  {
    localId: 'iOS-6B0F08E0-61B6-4009-BDBD-6D8A7769925C-1635150714053298',
    id: 'msg-vn-1-8AT5T0P0ET-1634524963617',
    conversationId: 'conv-vn-1-8AT5T0P0ET-1634524961033',
    sender: 'btncall_0000123452',
    createdAt: 1635146158214,
    state: 4,
    sequence: 1,
    type: 7,
    content: {
      participants: ['btncall_0000120493', 'btncall_0000123452'],
      creator: 'btncall_0000123452',
      groupName: 'btncall_0000120493',
    },
  },
];

export const realtimeConversation = {
  objectType: 0,
  objectChanges: [
    {
      id: 'conv-vn-1-WFHIO14TBS-1635589975318',
      name: 'app - web',
      isGroup: true,
      updatedAt: 1638866861310,
      creator: 'btncall_0000123452',
      created: 1635993700640,
      unreadCount: 2,
      participants: [
        {
          userId: 'btncall_0000123452',
          name: 'Nguyen Binh 02',
          avatar:
            'https://sandbox-citus.topenland.com/gateway/downloader/personal/24e9d3f9-539c-4d12-a5fe-ee4fe075daa0-1627115308781-ce73fb14-b285-47ac-a263-f1d1cf340da5.jpg',
        },
        {
          userId: 'btncall_0000120493',
          name: 'Binh03',
          avatar:
            'https://sandbox-citus.topenland.com/gateway/downloader/personal/805d50b6-435f-45b0-b76f-a9f1919512f9-1617355855183-0d65be4c-ca9a-4a40-b502-76208fd66abd.jpg',
        },
      ],
      lastMessage: {
        localId: null,
        id: 'msg-vn-1-BXCKDI4R11-1638446310301',
        conversationId: 'conv-vn-1-WFHIO14TBS-1635589975318',
        sender: 'btncall_0000120493',
        createdAt: 1638866861310,
        state: 3,
        sequence: 162,
        type: 1,
        content: {messageType: 1, text: '152', content: '152'},
      },
    },
  ],
  changeType: 0,
};

export const realtimeMessage = {
  objectType: 1,
  objectChanges: [
    {
      localId: 'iOS-879B4B86-849D-4445-BEAE-3ADC9DB3BF59-1638852104603241',
      id: 'msg-vn-1-BXCKDI4R11-1638446308979',
      conversationId: 'conv-vn-1-WFHIO14TBS-1635589975318',
      sender: 'btncall_0000120493',
      createdAt: 1638852104763,
      state: 4,
      sequence: 120,
      type: 1,
      content: {content: '111'},
    },
  ],
  changeType: 0,
};
