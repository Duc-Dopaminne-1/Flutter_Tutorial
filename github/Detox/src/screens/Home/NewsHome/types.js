import {ArticleDto} from '../../../api/graphql/generated/graphql';

export const defaultItemNews = {
  title: 'Quyết định dúng khi mua chung cư vẫn giữ được nhà đất',
  time: '5 giờ trước',
  bannerImage: null,
  isSmallStyle: true,
  articleType: 'Tin thị trường',
  onPress: () => {},
  customStyle: {},
};

export type NewsItem = typeof defaultItemNews;

export const defaultListNews = {listNews: [defaultItemNews]};

export type NewsProps = typeof defaultListNews;

export const mapAgentPropsUi = (data: ArticleDto) => {
  const newUi: NewsProps = {
    title: data.title,
    createdDatetime: data.createdDatetime,
    previewImageUrl: data.previewImageUrl,
    id: data.id,
    isSmallStyle: true,
    body: data.body,
    articleType: data?.articleType ?? '',
    slug: data?.slug ?? null,
  };
  return newUi;
};
