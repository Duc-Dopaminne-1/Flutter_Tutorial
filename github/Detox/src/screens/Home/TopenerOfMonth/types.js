import isEmpty from 'lodash/isEmpty';

import {
  PostContractStatus,
  SearchPropertyPostForRentInfoDto,
  SearchPropertyPostInfoDto,
  TopenerOfMonthDto,
} from '../../../api/graphql/generated/graphql';
import {
  APPROVAL_STATUS_ID,
  CommissionCurrencyUnit,
  getDirectionList,
} from '../../../assets/constants';
import {IMAGES} from '../../../assets/images';
import {translate} from '../../../assets/localize';
import {extractAddressData} from '../../../utils/DataProcessUtil';
import {dateToString} from '../../../utils/TimerCommon';
import PropertyPostUtils from '../../ManagePost/PropertyPostUtils';

export const defaultProperty = {
  isShowFollowButton: true,
  showBrokenInfo: false,
  isShowFollower: true,
  propertyInfo: {
    propertyPostId: 'propertyPostId',
    title: 'Bán căn hộ 2PN Chung cư Âu Cơ Tower',
    address: '01 Âu Cơ, Tân Thành, Tân Phú, Hồ Chí Minh.',
    images:
      'https://sandbox-new.topenland.com/gateway/downloader/sale/262391-pic0_new-c27e5294-f64b-4503-87c8-a8dba4ff1e72.jpg',
    isFollowed: false,
    propertyCode: '',
    numberOfBathrooms: 2,
    numberOfBedrooms: 2,
    direction: 'Tây bắc',
    buildingArea: 74.2,
    brokenName: 'Nguyễn Hưng Thịnh',
    brokenRating: 4,
    groupNameDescription: 'Rồng vàng',
    brokenPhone: '0928472261',
    brokenEmail: 'nguyenhungthinh16@gmail.com',
    brokenAvatar:
      'https://sandbox-new.topenland.com/gateway/downloader/sale/262391-pic0_new-c27e5294-f64b-4503-87c8-a8dba4ff1e72.jpg',
    commission: '1-2%',
    price: '1 tỷ 3',
  },
  style: [
    {marginBottom: 20, borderRadius: 5, backgroundColor: '#FFFFFF', flexWrap: 'nowrap'},
    {marginEnd: 16},
  ],
  isSmallStyle: true,
  propertyTypeName: 'apartment',
  buyRequestMode: false,
  rentCommission: 1,
  rentPrice: 1000000,
  isRented: false,
};

export type PropertyItemType = typeof defaultProperty;

const defaultTopener = {
  key: '1',
  agentId: 'agentId',
  rank: IMAGES.IC_TOPENER_RANK1,
  avatar: '',
  fullName: '',
  rating: '',
  groupNameDescription: '',
  rankType: 'rank1',
  groupName: 'Hưng Thịnh Corp',
  sellingTotal: 100,
  phone: '',
  email: '',
  properties: [defaultProperty],
};
type Topener = typeof defaultTopener;
export const defaultAgentProps = {
  topeners: [defaultTopener],
};

export type AgentProps = typeof defaultAgentProps;

export const mapAgentPropsUi = (data: TopenerOfMonthDto, formatPrice, index = 0) => {
  const getRankIcon = () => {
    if (index === 0) {
      return IMAGES.IC_TOPENER_RANK1;
    } else if (index === 1) {
      return IMAGES.IC_TOPENER_RANK2;
    } else {
      return IMAGES.IC_TOPENER_RANK3;
    }
  };
  const topener: Topener = {
    key: data.rankingName,
    agentId: data.agentId,
    rank: getRankIcon(),
    avatar: data.profilePhoto,
    rankType: data.rankingName,
    fullName: data.fullName,
    rating: data.rating,
    groupNameDescription: data.groupNameDescription,
    email: data.email,
    groupName: data.groupName,
    sellingTotal: data.sellingTotal,
    phone: data.phoneNumber,
    properties: data.propertyPostInfoDtos.map(property => {
      return mapProperty(property, formatPrice);
    }),
  };
  return topener;
};

export const formatDirection = direction => {
  return getDirectionList()?.find(item => {
    return item?.id?.toLowerCase() === direction?.toLowerCase();
  })?.name;
};

const parseImageJson = array => {
  let images = [];
  if (Array.isArray(array)) {
    images = array;
  } else {
    images = JSON.parse(array) ?? [];
  }
  return images;
};

const parseImageUrl = json => {
  const images = parseImageJson(json);
  if (images && images.length > 0) {
    const thumbnailImage = images.filter(e => e?.avatar || e?.checked);
    if (thumbnailImage.length === 0 || images.length === 1) {
      return images[0].url ?? images[0].uri ?? null;
    }
    return thumbnailImage[0]?.url ?? thumbnailImage[0]?.uri ?? null;
  }
  return null;
};

/**
 * Map property data response to UI object
 * @param {*} value Property post data
 * @param {*} formatPrice function format price int to string format
 * @param {*} hiddenHomeNumber (true/false) hidden/show home number of property
 * @returns mapping data object property post
 */
export function mapProperty(
  value: SearchPropertyPostInfoDto,
  formatPrice,
  hiddenHomeNumber = true,
) {
  const commission = PropertyPostUtils.parseCommission(
    value?.commission,
    value?.saleCommissionCurrencyUnitId,
    CommissionCurrencyUnit.PERCENTAGE,
  );

  const property: PropertyItem = {
    propertyPostId: value.propertyPostId,
    propertyTypeId: value?.propertyTypeId,
    images: parseImageUrl(value.images),
    isFollowed: value.isFollowed,
    isAgent: value?.sellerInfo?.isAgent,
    address: extractAddressData(value.propertyAddress, hiddenHomeNumber),
    title: value.postTitle,
    numberOfBedrooms: value.numberOfBedrooms,
    numberOfBathrooms: value.numberOfBathrooms,
    buildingArea: value.buildingArea ?? value.capetAreas,
    direction: formatDirection(value.direction),
    price: formatPrice(
      value.price,
      value?.unitOfMeasureId ?? value?.unitOfMeasure?.unitOfMeasureId,
    ),
    propertyCode: value?.propertyCode,
    commission,
    brokenAvatar: value.sellerInfo?.avatar,
    brokenName: value.sellerInfo?.fullName,
    brokenRank: value.sellerInfo?.agentRankingName,
    brokenRating: value.sellerInfo?.agentRating,
    brokenPhone: value.sellerInfo?.phoneNumber,
    brokenEmail: value.sellerInfo?.email,
    forRent: value.forRent ?? false,
    forSale: value.forSale ?? false,
    isSold: value.isSold ?? false,
    latitude: value.latitude,
    longitude: value.longitude,
    isRented: value?.isRented,
  };
  return property;
}

/**
 * Map rent property info to UI data
 * @param {*} value PropertyPost data
 * @param {*} formatPrice @function formatPrice format price function
 * @param {*} isForRent For rent property post
 * @returns mapping UI object rent property post
 * @see mapProperty
 */
export function mapPropertyForRent(value, formatPrice, isForRent = false) {
  const propertyPostForRent = isForRent ? value : value?.propertyPostForRentDto;
  if (!propertyPostForRent) {
    return null;
  }
  return {
    rentPrice: formatPrice(propertyPostForRent?.rentPrice),
    isRented: propertyPostForRent?.isRented,
    rentCommission: PropertyPostUtils.parseCommission(
      propertyPostForRent?.rentCommission,
      propertyPostForRent?.rentCommissionCurrencyUnitId,
    ),
    rentCommissionCurrencyUnitId: propertyPostForRent?.rentCommissionCurrencyUnitId,
    rentPeriod: propertyPostForRent?.rentPeriod,
  };
}

/**
 * Parse contract status key to string
 * @param {*} contractStatus Contract status value key
 * @returns translation string contract status
 */
export function mapPropertyContractStatus(contractStatus) {
  if (isEmpty(contractStatus)) return '';
  switch (contractStatus) {
    case PostContractStatus.Unsent:
      return translate('propertyPost.contractStatus.unsent');
    case PostContractStatus.Waittosign:
      return translate('propertyPost.contractStatus.waitToSign');
    case PostContractStatus.Waitforpay:
      return translate('propertyPost.contractStatus.waitForPay');
    case PostContractStatus.Haspaid:
      return translate('propertyPost.contractStatus.hasPaid');
    case PostContractStatus.Refuse:
      return translate('propertyPost.contractStatus.refuse');
    default:
      return '';
  }
}

export function mapSearchRentalProperty(value: SearchPropertyPostForRentInfoDto, formatPrice) {
  const propertyDetails: SearchPropertyPostInfoDto = {
    ...value?.searchPropertyPostInfoDto,
  };
  const rentCommission = PropertyPostUtils.parseCommission(
    value?.rentCommission,
    value?.rentCommissionCurrencyUnitId,
  );
  const commission = PropertyPostUtils.parseCommission(
    propertyDetails?.commission,
    propertyDetails?.saleCommissionCurrencyUnitId,
    CommissionCurrencyUnit.PERCENTAGE,
  );
  const property: PropertyItem = {
    propertyPostId: propertyDetails.propertyPostId,
    images: parseImageUrl(propertyDetails.images),
    isFollowed: propertyDetails.isFollowed,
    isAgent: propertyDetails?.sellerInfo?.isAgent,
    address: extractAddressData(propertyDetails.propertyAddress, true),
    title: propertyDetails.postTitle,
    numberOfBedrooms: propertyDetails.numberOfBedrooms,
    numberOfBathrooms: propertyDetails.numberOfBathrooms,
    buildingArea: propertyDetails.buildingArea ?? propertyDetails.capetAreas,
    direction: formatDirection(propertyDetails.direction),
    price: formatPrice(
      propertyDetails.price,
      propertyDetails?.unitOfMeasureId ?? propertyDetails?.unitOfMeasure?.unitOfMeasureId,
    ),
    commission,
    brokenAvatar: propertyDetails.sellerInfo?.avatar,
    brokenName: propertyDetails.sellerInfo?.fullName,
    brokenRank: propertyDetails.sellerInfo?.agentRankingName,
    brokenRating: propertyDetails.sellerInfo?.agentRating,
    brokenPhone: propertyDetails.sellerInfo?.phoneNumber,
    brokenEmail: propertyDetails.sellerInfo?.email,
    forRent: propertyDetails.forRent,
    forSale: propertyDetails.forSale,
    propertyCode: propertyDetails.propertyCode,
    rentCommission,
    rentPeriod: value?.rentPeriod,
    rentPrice: formatPrice(value?.rentPrice),
    latitude: propertyDetails.latitude,
    longitude: propertyDetails.longitude,
    isRented: propertyDetails?.isRented,
  };
  return property;
}

/**
 * Map guaranteed property (tin đăng ký gửi C2C)
 * @param {*} value PropertyPost data
 * @param {*} formatPrice @function formatPrice format price function
 * @param {*} isForRent For rent property post
 * @returns mapping UI object guaranteed property post
 * @see mapProperty
 */
export function mapPropertyC2CGuarantee(value, formatPrice, isCreatedUser, isForRent = false) {
  if (!value) return null;
  const postInfo = isForRent ? value?.searchPropertyPostInfoDto : value;
  const mappingProperty = mapProperty(postInfo, formatPrice, !isCreatedUser);
  const mappingPropertyForRent = mapPropertyForRent(value, formatPrice, isForRent);
  const property = {
    ...mappingProperty,
    ...mappingPropertyForRent,
    isGuaranteed: postInfo?.isGuaranteed,
    contractStatus: mapPropertyContractStatus(
      postInfo?.guaranteedPackage?.contractStatus ?? postInfo?.contractStatus,
    ),
    contractStatusValue: postInfo?.guaranteedPackage?.contractStatus ?? postInfo?.contractStatus,
    guaranteedPackageEndTime:
      postInfo?.guaranteedPackage?.guaranteedPackageEndTime ?? postInfo?.guaranteedPackageEndTime,
    postServiceType: postInfo?.postServiceType,
    requestedUpdatingReason: postInfo?.requestedUpdatingReason,
    isRequestedUpdate: postInfo?.propertyPostApprovalStatusId === APPROVAL_STATUS_ID.REQUEST,
    rejectedReason: postInfo?.rejectedReason,
    isRejected: postInfo?.propertyPostApprovalStatusId === APPROVAL_STATUS_ID.REJECTED,
    isCreatedUser: isCreatedUser,
    imagesSize: parseImageJson(postInfo?.images)?.length ?? 0,
    propertyPostApprovalStatusName: postInfo?.propertyPostApprovalStatusName ?? '',
    createdDatetime: dateToString(postInfo?.createdDatetime),
  };
  return property;
}
