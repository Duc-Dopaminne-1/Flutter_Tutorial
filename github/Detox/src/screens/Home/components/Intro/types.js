export type PageDataTypes = {
  chatDefault: ChatDefaultTypes,
  introData: IntroDataTypes,
};

export type ChatDefaultTypes = {
  botIcon: String,
  botName: String,
  botChatOne: String,
  userChatOne: String,
  botChatTwo: String,
  botChatThree: String,
  botChatFour1: String,
  botChatFourHyperLink: String,
  botChatFour2: String,
};

export type IntroDataTypes = Array<IntroDataItem>;

export type IntroDataItem = {
  title: String,
  image: int,
  description: String,
  chatData: Array<ChatDataTypes>,
};

export type ChatDataTypes = {
  servicesName: String,
  type: String,
  id: String,
};
