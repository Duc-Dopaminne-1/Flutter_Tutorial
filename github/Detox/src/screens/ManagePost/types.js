import {TextStyle} from 'react-native';

export type CommissionPieChartContainerProps = {
  totalAmount: Required<Number>,
  topenlandPercentage: Required<Number>,
  buyerPercentage: Required<Number>,
  sellerPercentage: Required<Number>,
  onPressConfigurePercentage: Function,
  hideCautionText: Boolean,
};

export type ConfigureCommissionSliderContainerProps = {
  totalAmount: Required<Number>,
  topenlandPercentage: Required<Number>,
  defaultBuyerPercentage: Number,
  defaultSellerPercentage: Number,
  onConfirm: Function,
  onClose: Function,
  onStartSliding: Function,
  onFinishedSliding: Function,
};

export type CommissionSliderProps = {
  title: String,
  input1Value: String,
  input1Title: String,
  input1TitleStyle: TextStyle,
  input1TitleDescription: String,
  input1TitleDescriptionStyle: TextStyle,
  input2Value: String,
  input2Title: String,
  input2TitleStyle: TextStyle,
  input2TitleDescription: String,
  input2TitleDescriptionStyle: TextStyle,
  maximumPercentage: Required<Number>,
  minimumPercentage: Number,
  percentage: Required<Number>,
  onValueChange: (values: Array<Number>) => {},
  onInput1Change: (values: Array<Number>) => {},
  onInput2Change: (values: Array<Number>) => {},
  themeColor: String,
  editable: Boolean,
  onSliding: Function,
  onSlideFinished: (values: Array<Number>) => {},
  overrideTrackstyle: ViewStyle,
};
