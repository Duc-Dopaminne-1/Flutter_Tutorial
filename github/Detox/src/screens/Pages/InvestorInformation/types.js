import {InvestorAddressInfoDto} from '../../../api/graphql/generated/graphql';

export type InvestorDetailTypes = {
  introduction: {
    logo: String,
    name: String,
    address: InvestorAddressInfoDto,
    website: String,
    phone: String,
    dateOfEstablishment: String,
    charterCapital: String,
    areas: String,
    introduce: String,
  },
  activityImages: String,
};
