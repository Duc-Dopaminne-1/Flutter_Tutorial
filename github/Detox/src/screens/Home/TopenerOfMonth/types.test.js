import {mapProperty} from './types';

export const mapPropertySample = () => {
  const data = {
    propertyPostId: 'ea7267e3-21f9-482c-8a9f-0644072d7b75',
    images:
      '[{"url": "https://sandbox-new.topenland.com/gateway/downloader/sale/260031-pic0_new-5b11826f-bc72-4d6f-a07b-87e57d2e3cba.jpg"}, {"url": "https://sandbox-new.topenland.com/gateway/downloader/sale/260031-pic1_new-a64312a8-7915-4bd3-8beb-c1ac7303d0c6.jpg"}, {"url": "https://sandbox-new.topenland.com/gateway/downloader/sale/260031-pic6_new-d2848083-6fdb-4ce3-9fe0-0f3804228d6e.jpg"}, {"url": "https://sandbox-new.topenland.com/gateway/downloader/sale/260031-pic8_new-d5bcb6c5-2f3d-4e31-af08-d5b2a52ebd57.jpg"}, {"url": "https://sandbox-new.topenland.com/gateway/downloader/sale/260031-pic11_new-4033a38b-d69e-4abb-bed2-4c55995d90fb.jpg"}]',
    price: 2650000000,
    postTitle: 'Bán căn chung cư Miếu nỗi, diện tích sàn 53 m2. 1 phòng ngủ, tầng cao',
    unitOfMeasureId: '2e1b0f25-2772-4d71-aca1-9bd0c8fb9a74',
    propertyTypeId: '2e2b0611-e1fc-4406-b13f-8afafff1c675',
    propertyAddress: {
      homeAddress: '01',
      streetName: 'Vũ Huy Tấn',
      wardName: 'Phường 13',
      districtName: 'Bình Thạnh',
      cityName: 'Hồ Chí Minh',
      __typename: 'PropertyAddressInfo',
    },
    forRent: true,
    forSale: true,
    isSold: false,
    isRented: false,
    numberOfBedrooms: 2,
    numberOfBathrooms: 2,
    buildingArea: 53,
    totalSiteArea: null,
    capetAreas: null,
    commission: 1,
    isCreatedByAgent: true,
    isFollowed: false,
    direction: 'east',
    sellerInfo: {
      sellerId: '377dc01c-72cd-4cf8-8a24-8def5e71c16e',
      fullName: 'Võ Khắc Phúc',
      avatar:
        'https://sandbox-new.topenland.com/gateway/downloader/personal/image-5d2c59c7-c46b-4102-81e4-d8425ee6f6ff.png',
      email: 'baongn02@nomail.com',
      phoneNumber: '0349679120',
      isAgent: true,
      agentRating: 4.7,
      agentRankingName: 'rank5',
      agentGroupDescription: 'Rồng Vàng 1',
    },
    __typename: 'SearchPropertyPostInfoDto',
  };
  return mapProperty(data, () => data.price);
};

it('map property', () => {
  expect(mapPropertySample()).toMatchSnapshot();
});
