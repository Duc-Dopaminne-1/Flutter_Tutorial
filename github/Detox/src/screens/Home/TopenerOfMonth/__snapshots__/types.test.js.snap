// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`map property 1`] = `
Object {
  "address": "Vũ Huy Tấn, Phường 13, Bình Thạnh, Hồ Chí Minh.",
  "brokenAvatar": "https://sandbox-new.topenland.com/gateway/downloader/personal/image-5d2c59c7-c46b-4102-81e4-d8425ee6f6ff.png",
  "brokenEmail": "baongn02@nomail.com",
  "brokenName": "Võ Khắc Phúc",
  "brokenPhone": "0349679120",
  "brokenRank": "rank5",
  "brokenRating": 4.7,
  "buildingArea": 53,
  "commission": "1%",
  "direction": "Đông",
  "forRent": true,
  "forSale": true,
  "images": "https://sandbox-new.topenland.com/gateway/downloader/sale/260031-pic0_new-5b11826f-bc72-4d6f-a07b-87e57d2e3cba.jpg",
  "isAgent": true,
  "isFollowed": false,
  "isRented": false,
  "isSold": false,
  "latitude": undefined,
  "longitude": undefined,
  "numberOfBathrooms": 2,
  "numberOfBedrooms": 2,
  "price": 2650000000,
  "propertyCode": undefined,
  "propertyPostId": "ea7267e3-21f9-482c-8a9f-0644072d7b75",
  "propertyTypeId": "2e2b0611-e1fc-4406-b13f-8afafff1c675",
  "title": "Bán căn chung cư Miếu nỗi, diện tích sàn 53 m2. 1 phòng ngủ, tầng cao",
}
`;
